src/lib/metrics/digitalOcean.ts

```diff
@@ -0,0 +1,94 @@
+import fetch from 'cross-fetch';
+import { capture } from '@snapshot-labs/snapshot-sentry';
+
+const BASE_URL = 'https://api.digitalocean.com/v2';
+const TTL = 60 * 1000; // 1 minute
+
+export default class DigitalOcean {
+  key?: string;
+  cache: Record<string, any> = {};
+  expirationTime = 0;
+
+  constructor() {
+    this.key = process.env.DIGITAL_OCEAN_TOKEN;
+
+    if (!this.key) {
+      throw new Error('[metrics:digital-ocean] Digital Ocean token is missing');
+    }
+  }
+
+  async apps() {
+    try {
+      const results = await this.#fetch(`apps`);
+      const apps: Record<string, any> = {};
+
+      for (const app of results.apps) {
+        apps[app.spec.name] = {
+          tier: app.spec.services[0].instance_size_slug,
+          instance_count: app.spec.services[0].instance_count,
+          region: app.region.slug,
+          country: app.region.flag,
+          last_active_deploy: app.last_deployment_active_at
+        };
+      }
+
+      return apps;
+    } catch (e: any) {
+      capture(e);
+      return false;
+    }
+  }
+
+  async appInstances() {
+    try {
+      const results = await this.#fetch(`apps/tiers/instance_sizes`);
+      const instances: Record<string, any> = {};
+
+      for (const datum of results.instance_sizes) {
+        instances[datum.slug] = datum;
+      }
+
+      return instances;
+    } catch (e: any) {
+      capture(e);
+      return false;
+    }
+  }
+
+  async balances() {
+    try {
+      return await this.#fetch(`customers/my/balance`);
+    } catch (e: any) {
+      capture(e);
+      return false;
+    }
+  }
+
+  async #fetch(path: string) {
+    const cacheKey = path;
+    const cachedEntry = this.cache[cacheKey];
+    const now = Date.now();
+    if (cachedEntry && this.expirationTime > now) {
+      return cachedEntry;
+    }
+
+    if (this.expirationTime < now) {
+      this.cache = {};
```

Comment:
Is it correct that we have one variable `expirationTime` for all cached items?

Reply:
Yes